core/src/commonMain/kotlin/org/jetbrains/litmuskt/Utils.kt

```diff
@@ -27,3 +27,25 @@ expect fun cpuCount(): Int
 
 @Suppress("UNCHECKED_CAST")
 fun <S> TypedArray(size: Int, init: (Int) -> S): Array<S> = Array<Any?>(size, init) as Array<S>
+
+fun <S> Array<S>.view(range: IntRange): Sequence<S> = sequence {
+    for (i in range) yield(this@view[i])
+}
+
+/**
+ * Split a range into [n] parts of equal (+/- 1) length.
+ */
+fun IntRange.splitEqual(n: Int): List<IntRange> {
+    val size = endInclusive - start + 1
+    val len = size / n // base length of each sub-range
+    val remainder = size % n
+    val delim = start + (len + 1) * remainder // delimiter between lengths (l+1) and l
+    return List(n) { i ->
+        if (i < remainder) {
+            (start + i * (len + 1))..<(start + (i + 1) * (len + 1))
```

Comment:
Perhaps adding spaces around `..<` will improve readability

Reply:
No spaces is (unfortunately) official codestyle